feat: Making Quartz available offline by making it a PWA (#465)
jackyzha0/quartz: docs/features/offline access.md

> ---

Reviewer: ```suggestion
---
title: "Offline Access (PWA)"
```

tags:
  - plugin/emitter
---

> This allows your Website to be accesible offline and be installed as an app. You can use it by adding `Plugin.Offline(),` to the `emitters` in `quartz.config.ts`

Reviewer: ```suggestion
This plugin allows your website to be accessible offline and be installed as an app. You can use it by adding `Plugin.Offline(),` to the `emitters` in `quartz.config.ts`
```

## Offline Capability

Whenever you visit a page it gets cached for offline use. Depending on the kind of content, the process for caching is diffent:

> - **Pages** (HTML, your converted Markdown files): Quartz first tries to get them over the Network. If that fails is goes back to the Cache

Reviewer: ```suggestion
- **Pages** (HTML, your converted Markdown files): Quartz first tries to get them over the Network. If that fails, your browser attempts to fetch it from the cache.
```

> - **Static Resources** (Fonts, CSS Styling, JavaScript): Quartz gets them from cache and updates the cache over the Network in the background. Next time the page loads this new cache is being used

Reviewer: ```suggestion
- **Static Resources** (Fonts, CSS Styling, JavaScript): Quartz uses cached resources by default and updates the cache over the network in the background.
```

> - **Images**: Those are getting saved once and then served from Cache. The amount of images is limited to 60 Images and are valid for 30 Days

Reviewer: ```suggestion
- **Images**: Images are saved once and then served from cache. Quartz uses a limited cache of 60 images and images remain in the cache for 30 days
```

You can edit the fallback page by changing the `offline.md` file in the root of your `content` directory

> ## App

Reviewer: ```suggestion
## Progressive Web App (PWA)
```

Progressive Web Apps can have [many properties](https://developer.mozilla.org/en-US/docs/Web/Manifest). We're only going to mention the ones Quartz supports by default, however you can edit the offline plugins file to add more in case required.

> ### Properties taken from configuration

Reviewer: ```suggestion
```

- **icons**: the `icon.svg` file in the `quartz/static` directory is used for all the icons. This makes it easier to scale the image since you don't need to provide an png for every size

> - **name** and **short_name**: For this we use the `pageTitle` you configured in `quartz.config.ts`

Reviewer: ```suggestion
- **name**, **short_name**: Uses the `pageTitle` configured in `quartz.config.ts`
```

> - **description**: For this we use the `description` you configured in `quartz.config.ts`

Reviewer: ```suggestion
- **description**: Uses the `description` configured in `quartz.config.ts`
```

> - **background_color** and **theme_color**: For this we use the `lightMode.light` color you configured in `quartz.config.ts`. However many browsers don't use that functionality

Reviewer: ```suggestion
- **background_color**, **theme_color**: Uses the `lightMode.light` color configured in `quartz.config.ts`.
```

> - **start_url**: For this we use the `baseUrl` you configured in `quartz.config.ts`

Reviewer: ```suggestion
- **start_url**: Uses the `baseUrl` configured in `quartz.config.ts`
```
